app/models/product.rb

```diff
@@ -20,8 +20,16 @@ class Product < ApplicationRecord
            through: :extension_products_associations,
            source: :extension
 
+  has_many :mirrored_extensions, -> { mirrored },
+    through: :extension_products_associations,
+    source: :extension
+
   enum product_type: { base: 'base', module: 'module', extension: 'extension' }
```

Comment:
Pros:
1. Copypasta from glue :grimacing:;
2. Easy to read;
3. Easy to add/remove types from enum.

Cons:
1. Takes more bytes on disk!